trace/generator/plugins/org.eclipse.gemoc.trace.gemoc/src/org/eclipse/gemoc/trace/gemoc/traceaddon/GenericTraceConstructor.java

```diff
@@ -83,10 +89,11 @@ public GenericTraceConstructor(Resource executedModel, Resource traceResource, M
 
 	private boolean addNewObjectToStateIfDynamic(EObject object, GenericState state) {
 		final EClass c = object.eClass();
-		final List<EStructuralFeature> mutableProperties = c.getEAllStructuralFeatures().stream()
-				.filter(p -> DynamicAnnotationHelper.isDynamic(p))
+		List<MutableField> fields = fieldExtractor.extractMutableField(object);
+		final List<EStructuralFeature> mutableProperties = fields.stream()
+				.map(f -> f.getMutableProperty())
 				.collect(Collectors.toList());
-		if (DynamicAnnotationHelper.isDynamic(object.eClass()) || !mutableProperties.isEmpty()) {
+		if (!fields.isEmpty()) {
```

Comment:
Yes I agree for the conditional :)

For IMututableFieldExtractor we can rename it to something like IDynamicPartAccessor and add into` def boolean isDynamic(EModelElement o) `

Reply:
(sorry for the delay)

OK with `IDynamicPartAccessor`!

Since with ALE you do not have any language-specific dynamic EClasses, we will have to call `isDynamic(object)` instead of `isDynamic(object.eClass())`, right?

Reply:
I am not sure what you mean by "language-specific dynamic EClasses".
Is it a class like Token?

Reply:
Not only. Let's say we have an `FSM` EClass in the AS, in which you add `currentState` using ALE. In such case, there will never exists a proper `FSM` dynamic EClass containing a `currentState` EReference, right? You will only manipulate at runtime a static object instance of the static `FSM` EClass of the AS, along with a generic companion dynamic object created by ALE containing the dynamic state of the object.

Reply:
Yes you are right for ALE

My question was more about isDynamic(object). What does it compute?
My understanding is it returns true if the class is declared in .ale or if the EClass is reopened to add new features.

Reply:
Yes, that would work! Another option would be to return `true` if a companion dynamic object exists for the static object given as parameter.